esmvaltool/diag_scripts/aerosols/aero_utils.py

```diff
@@ -0,0 +1,181 @@
+"""Part of the ESMValTool Aerosol diagnostics.
+
+This module contains utility functions commonly used by aerosol
+assessment routines.
+"""
+
+import iris
+import numpy as np
+
+
+class AeroAnsError(Exception):
+
+    """Exception class for errors raised when model data is checked in the
+    extract_pt module.
+    """
+
+
+def add_bounds(cube):
+    """Add bounds to a cubes latitude and longitude coordinates.
+
+    Parameters
+    ----------
+    cube : Iris cube
+        Iris cube with latitude and longitude coordinates.
+
+    Returns
+    -------
+    cube : cube
+        Iris cube with bounds added to the latitude and longitude coordinates.
+    """
+
+    if not cube.coord('latitude').has_bounds():
+        cube.coord('latitude').guess_bounds()
+    if not cube.coord('longitude').has_bounds():
+        cube.coord('longitude').guess_bounds()
+
+    return cube
+
+
+def extract_pt(icube, pt_lat, pt_lon, **kwargs):
+    """Extracts given location(s) (3-D) from a cube.
+
+    Method
+    ------
+    Uses Iris module Analysis.Interpolate to extract values
+    first based on horizontal coordinates and later based on
+    height, if specified.
+
+    If height ('altitude') is requested, checks if cube heights
+    include orography, i.e. HybridHeights have been derived.
+
+    Parameters
+    ----------
+    icube : Iris cube
+    pt_lat, pt_lon : latitude and longitude coordinates of desired points
+    kwargs: (optional)
+        height  : Altitude (above geoid) of point.
+        level   : Model/ pseudo/ tile number, default = all.
+        nearest : Boolean. Specify whether to use 'nearest neighbour', instead
+                  of 'linear' method while extracting data. Default is False.
+
+    Returns
+    -------
+    data_out : List
+        List of single point values, corresponding to each point specified
+
+    Raises
+    ------
+    AeroAnsError : If the numbers of latitude and longitude points are
+        mismatched. OR if both and level and height are passed as kwargs.
+        OR if the cude contains a time coordinate. OR if a pseudo level
+        coordinate is requested, but not present in the cube. OR if the numbers
+        of latitude/longitude and height points are mismatched. OR if height
+        is requested but the cube does not contain and altitude coordinate.
+    """
```

Comment:
very nice and descriptive raiser section :+1:

Reply:
:)